diff --git a/web-console/src/lib/services/manager/models/ColumnType.ts b/web-console/src/lib/services/manager/models/ColumnType.ts
index 8484d2bea..953d53760 100644
--- a/web-console/src/lib/services/manager/models/ColumnType.ts
+++ b/web-console/src/lib/services/manager/models/ColumnType.ts
@@ -57,5 +57,5 @@ export type ColumnType = {
    * - `DECIMAL(1,2)` sets scale to `2`.
    */
   scale?: number | null
-  type?: SqlType
+  type: SqlType
 }
diff --git a/web-console/src/lib/services/manager/models/ConnectorConfig.ts b/web-console/src/lib/services/manager/models/ConnectorConfig.ts
index 1e6fdd524..9c994ea19 100644
--- a/web-console/src/lib/services/manager/models/ConnectorConfig.ts
+++ b/web-console/src/lib/services/manager/models/ConnectorConfig.ts
@@ -9,7 +9,7 @@ import type { TransportConfig } from './TransportConfig'
  * A data connector's configuration
  */
 export type ConnectorConfig = OutputBufferConfig & {
-  format?: FormatConfig | null
+  format: FormatConfig
   /**
    * Backpressure threshold.
    *
diff --git a/web-console/src/lib/services/manager/models/TransportConfig.ts b/web-console/src/lib/services/manager/models/TransportConfig.ts
index a44285d34..774e36089 100644
--- a/web-console/src/lib/services/manager/models/TransportConfig.ts
+++ b/web-console/src/lib/services/manager/models/TransportConfig.ts
@@ -18,41 +18,47 @@ import type { UrlInputConfig } from './UrlInputConfig'
 export type TransportConfig =
   | {
       config: FileInputConfig
-      name: TransportConfig.name
+      name: TransportConfig.name.FILE_INPUT
     }
   | {
       config: FileOutputConfig
-      name: TransportConfig.name
+      name: TransportConfig.name.FILE_OUTPUT
     }
   | {
       config: KafkaInputConfig
-      name: TransportConfig.name
+      name: TransportConfig.name.KAFKA_INPUT
     }
   | {
       config: KafkaOutputConfig
-      name: TransportConfig.name
+      name: TransportConfig.name.KAFKA_OUTPUT
     }
   | {
       config: UrlInputConfig
-      name: TransportConfig.name
+      name: TransportConfig.name.URL_INPUT
     }
   | {
       config: S3InputConfig
-      name: TransportConfig.name
+      name: TransportConfig.name.S3_INPUT
     }
   | {
       config: DeltaTableReaderConfig
-      name: TransportConfig.name
+      name: TransportConfig.name.DELTA_TABLE_INPUT
     }
   | {
       config: DeltaTableWriterConfig
-      name: TransportConfig.name
-    }
-  | {
-      name: TransportConfig.name
+      name: TransportConfig.name.DELTA_TABLE_OUTPUT
     }
 export namespace TransportConfig {
   export enum name {
-    FILE_INPUT = 'file_input'
+    FILE_INPUT = 'file_input',
+    FILE_OUTPUT = 'file_output',
+    KAFKA_INPUT = 'kafka_input',
+    KAFKA_OUTPUT = 'kafka_output',
+    URL_INPUT = 'url_input',
+    S3_INPUT = 's3_input',
+    DELTA_TABLE_INPUT = 'delta_table_input',
+    DELTA_TABLE_OUTPUT = 'delta_table_output',
+    HTTP_INPUT = 'http_input',
+    HTTP_OUTPUT = 'http_output'
   }
 }
